pkg/kopia/command/helpers.go

```diff
@@ -0,0 +1,223 @@
+		utils.GetEnvAsIntOrDefault(kopia.DataStoreGeneralMetadataCacheSizeMBVarName, kopia.DefaultDataStoreGeneralMetadataCacheSizeMB)
+}
+
+// GetCacheSizeSettingsForRestore returns the feature setting cache size values to be used
+// for initializing repositories that will be performing restore workloads
+func GetCacheSizeSettingsForRestore() (contentCacheMB, metadataCacheMB int) {
+	return utils.GetEnvAsIntOrDefault(kopia.DataStoreRestoreContentCacheSizeMBVarName, kopia.DefaultDataStoreRestoreContentCacheSizeMB),
+		utils.GetEnvAsIntOrDefault(kopia.DataStoreRestoreMetadataCacheSizeMBVarName, kopia.DefaultDataStoreRestoreMetadataCacheSizeMB)
+}
+
+// SnapshotCreateInfo is a container for data that can be parsed from the output of
+// `kopia snapshot create`.
+type SnapshotCreateInfo struct {
+	SnapshotID string
+	RootID     string
+	Stats      *SnapshotCreateStats
+}
+
+// ParseSnapshotCreateOutput parses the output of a snapshot create command into
+// a new SnapshotCreateInfo struct and returns its pointer. The Stats field may be nil
+// if the stats were unable to be parsed. The root ID and snapshot ID are fetched from
+// structured stdout and stats are parsed from stderr output.
+func ParseSnapshotCreateOutput(snapCreateStdoutOutput, snapCreateStderrOutput string) (*SnapshotCreateInfo, error) {
+	snapID, rootID, err := SnapshotInfoFromSnapshotCreateOutput(snapCreateStdoutOutput)
+	if err != nil {
+		return nil, err
+	}
+
+	return &SnapshotCreateInfo{
+		SnapshotID: snapID,
+		RootID:     rootID,
+		Stats:      SnapshotStatsFromSnapshotCreate(snapCreateStderrOutput),
+	}, nil
+}
+
+// SnapshotCreateStats is a container for stats parsed from the output of a `kopia
+// snapshot create` command.
+type SnapshotCreateStats struct {
+	FilesHashed   int64
+	SizeHashedB   int64
+	FilesCached   int64
+	SizeCachedB   int64
+	SizeUploadedB int64
+}
+
+// SnapshotStatsFromSnapshotCreate parses the output of a kopia snapshot
+// create execution for a log of the stats for that execution.
+func SnapshotStatsFromSnapshotCreate(snapCreateStderrOutput string) (stats *SnapshotCreateStats) {
+	if snapCreateStderrOutput == "" {
+		return nil
+	}
+	logs := regexp.MustCompile("[\r\n]").Split(snapCreateStderrOutput, -1)
+
+	// Match a pattern starting with "*" (signifying upload finished), and containing
+	// the repeated pattern "<\d+> <type> (<humanized size base 10>),",
+	// where <type> is "hashed", "cached", and "uploaded".
+	// Example input:
+	// 	 * 0 hashing, 1 hashed (2 B), 3 cached (40 KB), uploaded 6.7 GB, estimated 1092.3 MB (100.0%) 0s left
+	// Expected output:
+	// SnapshotCreateStats{
+	// 		filesHashed:  1,
+	// 		sizeHashedB: 2,
+	// 		filesCached:  3,
+	// 		sizeCachedB: 40000,
+	// 		sizeUploadedB: 6700000000,
+	// }, nil
+	pattern := regexp.MustCompile(`\*.+[^\d](\d+) hashed \(([^\)]+)\), (\d+) cached \(([^\)]+)\), uploaded ([^\)]+),.+`)
```

Comment:
Should we declare regex as const?

Reply:
regex string can be a const and `pattern` could be a var defined once in the function. I believe that way it can be compiled once and reused each time the function is called